linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/GenIncOptimizer.scala

```diff
@@ -115,17 +102,22 @@ abstract class GenIncOptimizer private[optimizer] (config: CommonPhaseConfig) {
       }
 
       val newLinkedClasses = for (linkedClass <- unit.classDefs) yield {
-        def defs(container: Option[MethodContainer]) =
-          container.fold[List[Versioned[MethodDef]]](Nil)(_.optimizedDefs)
-
         val encodedName = linkedClass.encodedName
-        val memberNamespace =
-          if (linkedClass.kind == ClassKind.Interface) getDefaults(encodedName)
-          else getClass(encodedName)
+        val staticLikeContainers = staticLikes(encodedName)
+
+        val publicContainer = classes.get(encodedName).getOrElse {
+          staticLikeContainers(MemberNamespace.Public.ordinal)
+        }
+
+        val newMethods = for (m <- linkedClass.methods) yield {
+          val namespace = m.value.flags.namespace
+          val container =
+            if (namespace == MemberNamespace.Public) publicContainer
+            else staticLikeContainers(namespace.ordinal)
+          container.methods(m.value.encodedName).optimizedMethodDef
+        }
```

Comment:
I'm not sure I understand why this is now necessary. Couldn't we just take all methods in all containers?

Something like:

`classes.getOrElse(encodedName, empty) ++ statikLikeContainers.flatMap(_.defs)`

?

Reply:
It's necessary to preserve the order of methods as found in the .sjsir, which itself is necessary to preserve stability of the output.